### ISSUE TITLE
Additional 'description' field in application.yaml

### ISSUE THREAD

**Issue by andye2004** [argoproj/argo-cd#3040]
# Summary

Add a new description field to application.yaml

# Motivation

When looking at the applications tab in UI we sometimes have the same application deployed multiple times into different namespaces and the ability to add a description to the application and have it visible in application tab would really help understanding.

**Comment by jannfis** [comment]
Are you looking for a longer descriptive text? Or is it maybe something that could be solved using labels?

**Comment by andye2004** [comment]
@jannfis, Good question, I did think about using labels as it's visible on the tiled view of the application tab, although not on the list view. A few of things put me off this idea though:-

- Labels are name value pairs that are associated with a k8s deployment but what I'm looking for here is a metadata field at the argo application level.
- We already use quite a few labels for everything deployed into our cluster and separating genuine labels from something that is just a comment is less intuitive than just looking directly at a description field.
- Once a label is attached, even though I have the ability to delete the label in the edit screen and I click save, the label is not deleted (I probably should have raised this as a possible bug but I _thought_ it _may_ be by design)
- Using labels just didn't 'feel right'

Also, I noticed that there was an additional field called Info, again name/value pair on the edit screen of an application after it had been created and I also thought about using this but it doesn't appear on the tiled view of the application screen nor on the detailed view. And, again it's a name value pair so still feel right.

**Comment by alexmt** [comment]
Oh, the not deleted label is a bug.  @andye2004 could you please an issue for it?

**Comment by andye2004** [comment]
@alexmt, #3046 raised as requested.

**Comment by alexmt** [comment]
thank you @andye2004 !

**Comment by NawafSwe** [comment]
Hello @alexmt, is this issue still valid?

**Comment by bhuvan-somisetty** [comment]
Hi, I'd like to pick this up if it's still open.

To resolve the ambiguity from the earlier discussion, I'd propose adding a dedicated `description` string field to the Application spec (separate from `info`, since that's a name/value list and not quite suited for a free-text summary). It would show up as a tooltip/subtitle on the app tile and in the list view, so it's easy to tell apart apps with the same name across namespaces without relying on labels.

Happy to adjust the approach based on feedback — let me know if this sounds reasonable and I'll get started on a PR.

**Comment by Bryandero98** [comment]
I'd like to pick this up. @bhuvan-somisetty's proposal above (a dedicated `description` string field on the Application spec, separate from `info`, surfaced as a tooltip/subtitle on the app tile and list view) has been sitting unchallenged for about six weeks and directly resolves the ambiguity from the earlier discussion, so I'll build that unless a maintainer prefers a different shape.

Scope, so it's clear upfront: a new optional `Description` field on `ApplicationSpec` (`pkg/apis/application/v1alpha1/types.go`), regenerated via `make codegen` (deepcopy/protobuf/OpenAPI/CRD manifests), surfaced in the UI where the app name is shown, plus a docs update and tests per the contribution guidelines.

Could you assign this to me? I'll open a PR once it's ready.

**Comment by crenshaw-dev** [comment]
I think we need a clearer plan for how this field is meant to look and function in the UI and CLI. If there's a specific place or set of places in the UI we plan to use this, we probably want to limit the max number of characters so it sits nicely.

Even with a clearer design, I'm not sure the feature is a good idea. UI behavior is harder to test than backend behavior, and a feature used by a tiny subset of users is code that's likely to decay.

**Comment by bhuvan-somisetty** [comment]
Totally fair, and good questions. Here's roughly what I had in mind:

In the UI I'd put it as a small subtitle right under the app name, on both the tile and the list view, truncated with a tooltip for the full text on hover. To keep it from wrecking the layout I'd cap it server side (not just client side) at something like 64 characters. On the CLI side, it'd just be an extra column in `argo app list -o wide` and a line in `argo app get`, invisible if you never set it.

On whether it's worth the maintenance burden, I get the concern, but I think the actual surface area here is small. It's one optional string with no logic behind it, pretty much the same shape as an annotation, so there's not much to decay. I'd cover it with a unit test on the validation and one e2e test making sure it survives create/get/list.

Happy to write this up properly if the shape above sounds right, but if you still think it's not worth adding, no hard feelings, I'll drop it.

**Comment by crenshaw-dev** [comment]
Yeah, the backend surface is tiny. I'm only worried about the UI at some point breaking, but not enough people actually using the field for it to get caught and fixed.

I'm on the fence about whether it's worth it. I'll bring it up at the contributors meeting to see what the crew thinks.

**Comment by bhuvan-somisetty** [comment]
Sounds good, thanks for considering it and for taking it to the contributors meeting. I'll wait to hear what the crew thinks before doing anything further.

**Comment by Bryandero98** [comment]
Sounds good — I'll hold off on any further changes until the contributors meeting has a chance to weigh in.

**Comment by crenshaw-dev** [comment]
Consensus among maintainers was that this would not be worth the implementation/maintenance cost. Closing for now.

**Comment by bhuvan-somisetty** [comment]
No worries, thanks for taking it to the contributors meeting and for the honest feedback along the way. Makes sense given the maintenance tradeoff. Appreciate the discussion!
